src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prevNode;
+        private Node<T> nextNode;
+
+        Node(Node<T> prevNode, T value, Node<T> nextNode) {
+            this.prevNode = prevNode;
+            this.value = value;
+            this.nextNode = nextNode;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+        if (index == 0) {
+            final Node<T> newNode = new Node<>(null, value, head);
+            head.prevNode = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        checkIndex(index);
+        insertBefore(value, findNode(index));
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            linkLast(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        final Node<T> replacedNode = findNode(index);
+        T oldNodeValue = replacedNode.value;
+        replacedNode.value = value;
+        return oldNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlinkNode(findNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> lookup = head; lookup != null; lookup = lookup.nextNode) {
+                if (lookup.value == null) {
```

Comment:
```suggestion
                if (lookup.value == object || object != null && object.equals(lookup.value)) {
```
так не потрібно буде if else з двома циклами